src/components/Header/Header.tsx

```diff
@@ -0,0 +1,198 @@
+/* eslint-disable jsx-a11y/anchor-is-valid */
+import {
+  NavLink,
+  Link,
+  useLocation,
+  useSearchParams,
+} from 'react-router-dom';
+import {
+  useCallback,
+  useContext,
+  useEffect,
+  useState,
+} from 'react';
+import cn from 'classnames';
+import { LogoLink } from '../LogoLink';
+import { ProductsContext } from '../../context/ProductsContext';
+import { setDebounce } from '../../helpers/setDebounce';
+
+export const Header = () => {
+  const { pathname } = useLocation();
+  const [searchPlaceholder, setSearchPlaceholder] = useState('');
+  const [isTabCard, setIsTabCard] = useState(false);
+  const {
+    favourites,
+    cart,
+    setIsLoading,
+  } = useContext(ProductsContext);
+  const [searchParams, setSearchParams] = useSearchParams();
+  const [inputValue, setInputValue]
+    = useState(searchParams.get('query') || '');
+
+  const isLinkActive = (isActive: boolean) => {
+    return cn('header__nav-link', {
+      active: isActive,
+    });
+  };
+
+  const debounce = useCallback(setDebounce(), []);
+
+  const setSearch = (value: string) => {
+    if (value) {
+      searchParams.set('query', value);
+      searchParams.delete('page');
+    } else {
+      searchParams.delete('query');
+    }
+
+    setSearchParams(searchParams);
+    setIsLoading(false);
+  };
+
+  const onInputChanged = (event: React.ChangeEvent<HTMLInputElement>) => {
```

Comment:
pay attention to naming. why `onInputChanged` when it is just a handler?
```suggestion
  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
```